utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/source/UniversalKafkaSourceFactory.scala

```diff
@@ -248,8 +248,11 @@ class UniversalKafkaSourceFactory(
         NonEmptyList.one(CustomNodeError(nodeId.id, "Cannot generate test parameters: no runtime schema found", None))
       )
       .andThen { runtimeSchema =>
-        val universalSchemaSupport: UniversalSchemaSupport =
-          schemaSupportDispatcher.forSchemaType(runtimeSchema.schema.schemaType())
+        val universalSchemaSupport = runtimeSchema.schema match {
+          // For ad hoc tests we want to present the user with json editor when schemaless topic with Json content type was selected
+          case ContentTypesSchemas.schemaForJson => EmptySchemaJsonSupport
```

Comment:
Let's look a momement at this code from perspective of what user see (DDD). User see that they choose topic without schema. They have to pick content type for this topic. So from this perspective I expect to see in the code:
```
case TopicWithoutSchema(contentType = Json) => NoSchemaJsonSupport or SchemalessJsonSupport
```
First part of this pattern was hacked - we have to fix eventually to avoid some bugs around this subject (we already have some because of this hack). But let's avoid going deeper into this hack and fix at least second part of pattern matching.

LSS
Please rename `EmptySchemaJsonSupport` to something more closer to domain: `NoSchemaJsonSupport` or `SchemalessJsonSupport`

Reply:
It is better to use `ParsedSchemaSupport` interface. The logic behind `ParsedSchemaSupport` was to dispatch all logic that is different depending on what schema is associated with topic. Now we by-pass this dispatching logic. Following the original idea, it will be easy to show differences between topic with json schema and topic with json schema - it will be in one place.